runtime/tests/redirector/genericlauncher/genlauncher.c

```diff
@@ -211,7 +211,7 @@ int main(int argc, char ** argv, char ** envp)
 		} else if (0 == strcmp(argv[i], "-cp")) {
 			/* convert "-cp classpath" to "-Djava.class.path=classpath" */
 			i++; /* the next option is the classpath itself */
```

Comment:
While you're here, please validate that `i` is legal after this increment.

Reply:
See line 154. There is a pre-computation that ensures it is legal.

Reply:
Perhaps you meant line 164, but I agree it has ensured there's an argument to be paired with `-cp`.